compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -2552,6 +2554,41 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             _ => {}
         }
     }
+
+    fn suggest_derive(&self, err: &mut Diagnostic, trait_pred: ty::PolyTraitPredicate<'tcx>) {
+        let Some(diagnostic_name) = self.tcx.get_diagnostic_name(trait_pred.def_id()) else {
```

Comment:
I think you can combine this into the `match diagnostic_name` below.

Reply:
Do you mean code like the following?
```rust
        let can_derive = match self.tcx.get_diagnostic_name(trait_pred.def_id()) {
            Some(sym::Default) => !adt.is_enum(),
            Some(sym::PartialEq) | Some(sym::PartialOrd) => {
                let rhs_ty = trait_pred.skip_binder().trait_ref.substs.type_at(1);
                self_ty == rhs_ty
            }
            Some(sym::Eq)
            | Some(sym::Ord)
            | Some(sym::Clone)
            | Some(sym::Copy)
            | Some(sym::Hash)
            | Some(sym::Debug) => true,
            _ => false,
        };
```

But I used this `diagnostic_name` variable [here](https://github.com/rust-lang/rust/pull/95525/files#diff-aef5d12ac7ec2060165cf1db78128f941c789bb2fc90c7ed351049b0be4d435eR2585) and [here](https://github.com/rust-lang/rust/pull/95525/files#diff-aef5d12ac7ec2060165cf1db78128f941c789bb2fc90c7ed351049b0be4d435eR2587) .

Reply:
sorry, I understand now!